cmd/infra/powervs/create.go

```diff
@@ -1148,6 +1159,28 @@ func (infra *Infra) setupPowerVSDHCP(ctx context.Context, options *CreateInfraOp
 	return nil
 }
 
+// monitorDhcpStatus monitor DHCP status to reach either ACTIVE or ERROR status
+func monitorDhcpStatus(client *instance.IBMPIDhcpClient, infraID string, dhcpID string) (*models.DHCPServerDetail, bool, error) {
+	var err error
+	dhcpServer, err := client.Get(dhcpID)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if dhcpServer != nil {
+		log(infraID).Info("Waiting for DHCPServer to up", "id", *dhcpServer.ID, "status", *dhcpServer.Status)
+		if *dhcpServer.Status == dhcpServiceActiveState {
+			return dhcpServer, true, nil
+		}
+
+		if *dhcpServer.Status == dhcpServiceErrorState {
+			return dhcpServer, true, fmt.Errorf("dhcp service is in error state")
```

Comment:
Why/when this function return true/false? is not clear by the name.

Reply:
Updated the comment explaining the return values. ptal!

Reply:
>// returns an instance of DHCPServerDetail for further processing and true if reaches either ACTIVE or ERROR status

I see, thanks.  Then you use isReady to gather the value which makes me think the return value is evaluating if is ACTIVE or not, but that's not what's happening.

Should the function return true when isActive and so rename the variable?

Should we I'd rather break this down into two functions getDHCPServer() and isDHCPServerActive()?